manifests/install.pp

```diff
@@ -4,16 +4,19 @@
 #
 class consul::install {
 
-  if $::consul::data_dir {
-    file { $::consul::data_dir:
+  if ($::consul::data_dir) and ($::consul::install_method != 'docker') {
```

Comment:
Mostly because of the nature of consul+puppet itself makes it unnecessary.  In a production environment you should have 3/5/n masters so all info will replicate, and with a puppet config all checks and services will populate on first run so it's simply unnecessary to store that information beyond the life-cycle of the container (remember volumes will persists in upgrades, just not with you deliberately rm the container)